src/components/mermaidMap/MiniMap/MiniMap.js

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react'
+import maplibregl from 'maplibre-gl'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { lightBaseMap } from '../mapService'
+
+const MiniMapWrapper = styled.div`
+  position: absolute;
+  width: 200px;
+  height: 150px;
+  border: 2px solid white;
+  opacity: ${({ visible }) => (visible ? 1 : 0)};
```

Comment:
chose opacity instead of display to hide/unhide the minimap. Was having issues where the minimap would render before the wrapper and exceed the container. It was inconsistent - particularly if it was a new site and a pin was added right away. This was happening despite having the code below in the `useEffect`

```
if (!miniMapContainer.current || !mainMap) {
      return
    }
```

Reply:
oh how annoying. Can you add a inline comment for the opacity line explaining that?

Reply:
yes - will add a comment